### ISSUE TITLE
[Code] Add Continuous Integration pipeline (CI)

### ISSUE THREAD

**Issue by albincorreya** [MichaelSel/edoJS#12]
According to [JOSS guidelines](https://joss.readthedocs.io/en/latest/review_criteria.html#tests), it is highly encouraged to use [Continous Integration (CI)](https://en.wikipedia.org/wiki/Continuous_integration) tools for automating tests, publishing to package manager, etc. This can be very useful for easily maintaining the repository. However, these are not strictly mandatory according to JOSS guidelines.

Since the repo is on Github, I encourage you to use [Github Actions](https://docs.github.com/en/actions) for adding CI/CD to the repo. I think it can be set up quite easily by specifying a YAML configuration in the repository.

### Automated Tests (Related to #8) 

I can see there are some unit tests defined [here](https://github.com/MichaelSel/edoJS/blob/master/test/unit.js) in the repository. 

You could add an entry point to run your tests in the `package.json` file. Just replace [this line](https://github.com/MichaelSel/edoJS/blob/master/package.json#L7) with `"test": "mocha"`.

Why? - You could run the test with the command `npm run test`. This can be useful for both local development and in your CI pipeline.

### Pipeline to publish the library into NPM or similar (Optional) (related to #11 )

Add a step in your pipeline to publish `edo.js` library to NPM or any JS package managers. Here is a [guide](https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages) on how to do it with Github Actions.

Alternatively, you could also do the same thing from your local computer considering you're the only maintainer of the library.

## Why?

Once this setup is done,  you can easily release the new updates of the library once it is reviewed and merged into your master branch automatically. In addition, you can also run automated tests whenever a change is made in the repository.

A typical example workflow would be something like this,

`Merge PR -> Run Github actions (ie, Install project dependencies -> Run unit tests -> Build the library files to various JS specifications (minified, UMD, ES* style etc) -> Update version and publish to NPM)`

I hope it is helpful. Let me know if something is not clear!
